### PR TITLE
docs: split up ruleset docs

### PR DESCRIPTION
**Pull request by philsturgeon**
Rulesets were getting rather overloaded, and we had a looooot of advanced content in a "getting started" section. This breaks things up a little more gracefully, so we learn a bit about rulesets, then get stuck into nitty gritty later on.

### CODE REVIEW

File: docs/guides/1-workflows.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
-# Workflow
+# Workflows
 
 When and where should you use Spectral? It depends a lot how you are creating and managing your API description documents, but probably wherever they are being made.
 
-- Run [Spectral CLI](./cli.md) against design docs and get feedback very early on.
-- Run Spectral in [Stoplight Studio](https://stoplight.io/spectral/) automatically as you work, without switching to the CLI.
+- Run [Spectral CLI](2-cli.md) against design docs and get feedback very early on.
+- Run Spectral in [Stoplight Studio](https://stoplight.io/studio/) or [VS Code](https://github.com/stoplightio/vscode-spectral) automatically as you work, without switching to the CLI.
```
Comment (philsturgeon) [inline docs/guides/1-workflows.md:6]: Sneaking in links here even though its not a full announcement.

File: docs/guides/4-custom-rulesets.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
+# Custom Rulesets
```
Comment (philsturgeon) [inline docs/guides/4-custom-rulesets.md:1]: Lots of stuff in this file has simply been reordered, not removed.

File: docs/getting-started/rulesets.md [hunk 3]
```diff
@@ -1,369 +1,101 @@
...
-    recommended: true
-    then:
-      field: description
-      function: truthy
-```
-
-Spectral has a built-in set of functions such as `truthy` or `pattern`, which you can reference in your rules. Rules then target certain chunks of the JSON/YAML with the `given` keyword, which is a [JSONPath](http://jsonpath.com/) (actually, we use [JSONPath Plus](https://www.npmjs.com/package/jsonpath-plus)).
-
-The example above adds a single rule that looks at the root level "tags" objects children to make sure they all have a description property.
+A ruleset is a JSON or YAML file (often the file will be called `.spectral.yaml`), and there's two main parts.
```
Comment (nulltoken) [inline docs/getting-started/rulesets.md:9]: there are two main parts

File: docs/getting-started/rulesets.md [hunk 4]
```diff
@@ -1,369 +1,101 @@
...
-  "tags": [{
-    "name": "animals",
-    "description": "come in all shapes and sizes"
-  }]
-}
-```
-
-By default, Spectral processes each rule on resolved document with all $refs resolved.
-If you would like to have an original input supplied to your rule, you can place `resolved` property as follows:
+Rules make which might look a bit like this:
```
Comment (nulltoken) [inline docs/getting-started/rulesets.md:13]: I've got slight issues understanding this sentence.

Comment (nulltoken) [inline docs/getting-started/rulesets.md:13]:
```suggestion
Rules might look a bit like this:
```

File: docs/getting-started/rulesets.md [hunk 5]
```diff
@@ -1,369 +1,101 @@
...
-    foo: bar
-```
-
-## Formats
+### Formats
 
 Formats are an optional way to specify which API description formats a rule, or ruleset, is applicable to. Currently Spectral supports these formats:
 
-- `oas2` (this is OpenAPI v2.0 - formerly known as Swagger)
+- `oas2` (this is OpenAPI v2.0)
```
Comment (nulltoken) [inline docs/getting-started/rulesets.md:55]: @philsturgeon Do we really need the repeating `this is`?

File: docs/guides/4-custom-rulesets.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+```
+
+## Extending Rules
+
+When extending another ruleset, you can actually extend and modify rules it has declared by adding a rule to your own ruleset with the same name.
+
+```yaml
+extends: spectral:oas
+rules:
+  my-rule-name:
```
Comment (nulltoken) [inline docs/guides/4-custom-rulesets.md:91]: @philsturgeon This example doesn't really relate with the previous sentence

File: docs/getting-started/rulesets.md (hunk 4 above)
Comment (philsturgeon) [reply docs/getting-started/rulesets.md:13]: It was nonsense!

**Comment by philsturgeon** [comment]
Alright ready for another pair of eyes. @mnaumanali94 can you take a look at this? 

I got distracted sorting this out during my style guide article writing last week, gonna get back on it now.

**Comment by mnaumanali94** [comment]
@philsturgeon I'll go through this in a bit. Just finishing up an article on my side.

File: docs/guides/4-custom-rulesets.md (hunk 6 above)
Comment (nulltoken) [reply docs/guides/4-custom-rulesets.md:91]: image

@philsturgeon I may be missing something here. The doc says "You can actually extend and modify rules it has declared by adding a rule to your own ruleset with **the same name**.", but `my-rule-name` doesn't exist in the oas ruleset. Am I misunderstanding what this sample demonstrates?

Comment (philsturgeon) [reply docs/guides/4-custom-rulesets.md]: Fixed!

File: docs/guides/5-custom-functions.md [hunk 7]
```diff
@@ -189,117 +258,74 @@ module.exports = (targetVal, opts) => {
...
-- if you Spectral programmatically via JS API and your ruleset remains unchanged, all subsequent `spectral.run` calls will invoke custom functions with the same cache instance.
-As soon as you set a ruleset using `setRuleset` or `loadRuleset` method, each custom function will receive a new cache instance.
-- if you are a CLI user, the cache will never be invalidated during the timespan of process.
+- data / credentials infiltration,
+- data tampering,
+- running cpu-intensive tasks, i.e. crypto-mining.
 
-Please bear in mind that while you are welcome to store certain kind of data, using cache for exchanging information between subsequent function calls it strongly discouraged.
-Spectral does not guarantee any particular order of execution meaning the functions can be executed in random order, depending on the rules you have, and the document you lint.
+While the risk is relatively low, you should be careful about including **external rulesets** you are not in charge of, in particular these that leverage custom functions.
```
Comment (mnaumanali94) [inline docs/guides/5-custom-functions.md:319]:
```suggestion
While the risk is relatively low, you should be careful about including **external rulesets** you are not in charge of, in particular those that leverage custom functions.
```

File: docs/guides/5-custom-functions.md [hunk 8]
```diff
@@ -189,117 +258,74 @@ module.exports = (targetVal, opts) => {
...
+You are strongly encouraged to review the custom functions a given ruleset provides.
+What you should hunt for is:
+- obfuscated code,
+- calls untrusted external library,
```
Comment (mnaumanali94) [inline docs/guides/5-custom-functions.md:323]:
```suggestion
- calls to an untrusted external library,
```

File: docs/guides/5-custom-functions.md [hunk 9]
```diff
@@ -189,117 +258,74 @@ module.exports = (targetVal, opts) => {
...
-As such, when your custom function is susceptible to return more than one result, you have to specify a different `path`
-for each result.
+- Using the JavaScript API, so long as your ruleset remains unchanged, all subsequent `spectral.run()` calls will invoke custom functions with the same cache instance.
+As soon as you set a ruleset using `setRuleset()` or `loadRuleset()` method, each custom function will receive a new cache instance.
 
-Below a sample function that checks tags bear unique names.
+- Using the CLI, the cache will be invalidated when the process terminates.
 
-**my-ruleset.yaml**
+You can store any kind of data, using cache for exchanging information between subsequent function calls it strongly discouraged. Also, Spectral does not guarantee any particular order of execution meaning the functions can be executed in random order, depending on the rules you have, and the document you lint.
```
Comment (mnaumanali94) [inline docs/guides/5-custom-functions.md:288]:
```suggestion
You can store any kind of data, using cache for exchanging information between subsequent function calls is strongly discouraged. Also, Spectral does not guarantee any particular order of execution meaning the functions can be executed in random order, depending on the rules you have, and the document you lint.
```

**Pull request merged**